homeassistant/components/geniushub/switch.py

```diff
@@ -26,6 +35,33 @@ async def async_setup_platform(
         ]
     )
 
+    async def set_switch_override(call) -> None:
+        """Set the system mode."""
+        entity_id = call.data[ATTR_ENTITY_ID]
+
+        registry = await hass.helpers.entity_registry.async_get_registry()
+        registry_entry = registry.async_get(entity_id)
+
+        if registry_entry is None or registry_entry.platform != DOMAIN:
+            raise ValueError(f"'{entity_id}' is not a known {DOMAIN} entity")
+
+        if registry_entry.domain != "switch":
+            raise ValueError(f"'{entity_id}' is not an {DOMAIN} zone")
+
+        payload = {
+            "unique_id": registry_entry.unique_id,
+            "service": call.service,
+            "data": call.data,
+        }
+        async_dispatcher_send(hass, DOMAIN, payload)
+
+    hass.services.async_register(
+        DOMAIN,
+        SVC_SET_SWITCH_OVERRIDE,
+        set_switch_override,
+        schema=SET_SWITCH_OVERRIDE_SCHEMA,
+    )
```

Comment:
This should be implemented as an Entity Service.

See: <https://developers.home-assistant.io/docs/dev_101_services#entity-services>

Reply:
OK

Reply:
Not sure why this was resolved, as the requested change was not implemented.

Reply:
I moved the code into __init__.py which is what I thought was required.  

Should this be kept in switch.py? I'll move it back.

Reply:
It should be registered as an entity service, using `async_register_entity_service`

See the developer link above for an example.

Reply:
Done.

Reply:
No it isn't done. It is still not using an Entity Service as listed above now a couple of time. Could you please take a moment and read the developers link I have listed in my first response?

Reply:
Sorry - I was trying to follow the ambient style. The existing code does not use `async_register_entity_service`. I will, attempt to, refactor the code to follow the example.

Reply:
Not all code uses the new methods yet, however, more and more are migrated. We stopped accepting old style, or else the migration will never end.

Reply:
Understand. For reference, should I only refactor the switch entity or all three (switch, climate and binary_sensor)?
Maybe get the switch done first and raise new PRs for the others would be better?

I can see many components use `async_register_entity_service` so I should be able to find a good pattern to follow.

The example uses `platform.async_register_entity_service`. To be clear, I should be using `component.async_register_entity_service`. Please confirm.

Reply:
You should limit this PR purely to your added service.

Refactoring of other services should be done in a separate PR.

Reply:
OK - I hope this is right now. I see how the new methods actually simplifies the code, which is a good thing.
If this is correct now, I will take a look at moving the other parts of this component forward.